### ISSUE TITLE
🐛 Bug Report: After 1.5.7: Realtime relationship payload removed and nothing returned

### ISSUE THREAD

**Issue by tawfeks** [appwrite/appwrite#8599]
### 👟 Reproduction steps

In a chat app, there are messages related to chats.

When server updates a message, it sends realtime client update to that message with a relationship chat id to the chat opened with details.

**In 1.5.7 and before:**
Everything was good

**After 1.5.7:**
This has been added:
https://github.com/appwrite/appwrite/pull/8381

This removes relationships id/s and details from realtime payloads which breaks my app.

### 👍 Expected behavior

Returns relationship id/s in realtime payload.
Or
Return relationship payload as before (with full data)

### 👎 Actual Behavior

Returns nothing for realtime relationship payload

### 🎲 Appwrite version

Version 1.5.x

### 💻 Operating system

Linux

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by SafakB** [comment]
🎲 Tested Appwrite version
Version 1.5.10

I just encountered this problem. Although I added the ‘Any’ permission to both collections, the relationship does not appear in the realtime payload.

![image](https://github.com/user-attachments/assets/07614822-1429-437d-9938-b6af7bdda0c9)

**Comment by shprotx** [comment]
The same problem with 1.5.10. There are no relationship-fields in payload.

![Screenshot 2024-09-09 034807](https://github.com/user-attachments/assets/da5da494-c400-4428-8605-2835aaa201a9)

`payload = {
    creatorID=66523586002da49b6aaf, 
    date=1.725835124022E12, 
    $id=b3a32084-4a55-4da8-a960-984ecc44d9a3, 
    $createdAt=2024-09-08T22:38:29.048+00:00, 
    $updatedAt=2024-09-08T22:38:45.986+00:00, 
    $permissions=[read("user:66523586002da49b6aaf"), update("user:66523586002da49b6aaf"), 
delete("user:66523586002da49b6aaf")], 
    $databaseId=63e2b4503fa1bf5d1a5f, 
    $collectionId=6506cefb97fbf5092408
}`

**Comment by shprotx** [comment]
https://github.com/appwrite/appwrite/pull/8381 - reason

**Comment by LDY681** [comment]
It doesn't make sence as altering relationship attributes would trigger realtime events on the target collection, but altered values won't be returned in the event payload. There should be a consistency of behaviors in both senders and receivers.

**Comment by LDY681** [comment]
Where relationship data could be shown should be based on ownership and permission scope, like what you did with new Databases().getDocument(). This is a bad decision to remove all of them, please reconsider🙏 @byawitz @stnguyen90

**Comment by docimin** [comment]
Heyo!

Would like this feature to come back for the following reason (anything this path is applied):

I have 2 collections, called for example
- messages-contacts
- messages

Since my platform has a lot of options to go towards, each user has a "userdata" collection. Each "contact" is a user that I have messaged before, or at least want to message.
Opening a chat, or adding a contact in general requires the userdata to be retrieved, which can easily be done by a relationship.

Since everyone can read the userdata (public) data, the relationship would be no issue at all to be included and even so, I think you can't even read relationship data if you don't have access to the data the relationship is referring to AFAIK.

This saves a lot of API calls, especially with a lot of so called "contacts" are changing, as each contact requires an additional api call and for other people, maybe even more, depending on their setup.

**Comment by spencerby28** [comment]
I would agree that this should be optional on relationships. The realtime API will send events on updates of a relationship attribute, although none of its data is sent. Let's say you have a function that updates both the attribute with a relationship as well as a document's wholly owned attribute. Realtime sends two updates: one where the returned data is "unchanged" as it is triggered on the relationship update but no relationship data is returned. Then another when the core attribute to that document is updated. Poor design choice imo. A flag that allows relationships in realtime connections based on permissions would make the most sense.
